CMakeLists.txt

```diff
@@ -162,24 +160,29 @@ IF(NOT CYCLUS_DOC_ONLY)
     MESSAGE("-- Dependency Binary Hints (DEPS_BIN_HINTS): ${DEPS_BIN_HINTS}")
     MESSAGE("-- Dependency Library Hints (DEPS_LIB_HINTS): ${DEPS_LIB_HINTS}")
     MESSAGE("-- Dependency Include Hints (DEPS_INCLUDE_HINTS): ${DEPS_INCLUDE_HINTS}")
+    MESSAGE("CMAKE_MODULE_PATH: ${CMAKE_MODULE_PATH}")
+
+    # Search pkg-config utility first
+    find_package(PkgConfig REQUIRED)
+
 
     # Debian installs useful LibXML2 files to /usr/include/libxml2/libxml
     # libxml2 is required for relaxng schema validation
-    FIND_PACKAGE(LibXml2 ${DEPS_HINTS})
-    IF(NOT LIBXML2_LIBRARIES)
-        FIND_LIBRARY(LibXml2 REQUIRED ${DEPS_HINTS})
-    ENDIF()
+
+    FIND_PACKAGE(LibXml2 REQUIRED)
     ADD_DEFINITIONS(${LIBXML2_DEFINITIONS})
     SET(LIBS ${LIBS} ${LIBXML2_LIBRARIES})
-
-    # Find LibXML++ and dependencies
-    FIND_PACKAGE(LibXML++)
-    IF(NOT LibXML++_LIBRARIES)
-        FIND_LIBRARY(LibXML++ REQUIRED ${DEPS_HINTS})
-    ENDIF()
-    SET(LIBS ${LIBS} ${LibXML++_LIBRARIES})
-    message("-- LibXML++ Include Dir: ${LibXML++_INCLUDE_DIR}")
-
+    message("-- LibXML2 Include Dir: ${LIBXML2_INCLUDE_DIR}")
+
+    # Then use pkg-config for locate specific package
+    pkg_check_modules(LIBXMLXX IMPORTED_TARGET libxml++-4.0)
+    IF ( NOT LIBXMLXX_LIBRARIES )
+       pkg_check_modules(LIBXMLXX REQUIRED IMPORTED_TARGET libxml++-2.6)
+    ENDIF ( NOT LIBXMLXX_LIBRARIES )
```

Comment:
This is because Ubuntu packages only support up to v2.6 right now.  If we want that to be a path available to users, we need to support one or the other.

Reply:
That makes sense, but then in line 493 we set the `CPACK_DEBIAN_PACKAGE_DEPENDS` to `libxml++4.0-dev (>=4.0.0)`. Would that interfere with using either version of libxml++?

Reply:
Did we figure out if these two things conflict if a user has libxml++ 2.6?

Reply:
If making a debian package it means we are using apt get so it should be 2.6.  I updated that.